### ISSUE TITLE
Reproducing M2RNN State Tracking

### ISSUE THREAD

**Issue by smdrnks** [open-lm-engine/lm-engine#411]
Hi! Cool training library and interesting work on M2RNN!

I am trying to reproduce the S3 state tracking results from M2RNN. I tried with the hyperparameters from Deltaproduct: 
2 million training samples at 128 context size for 100 epochs, testing extrapolation on 500k sequences of length 512. Using Deltaproduct with num_housholders=2 and GRU or LSTM the model quickly learns the task. So far I did not have success with M2RNN (with both a multi-value or deltaproduct matched setup). I tried with gradient clipping and identity init like you describe in the paper and ran a small learning rate + random seed sweep.

Could you maybe share the exact hyper parameters you have used? Would be greatly appreciated :) Also please let me know if this is the wrong place to discuss this.

**Comment by mayank31398** [comment]
Hey, yes!
Both M2RNN and traditional RNN is hard to train for the S3 task. I spent a few weeks with a (few 100s of H100s) trying to train the model using it.

**Comment by mayank31398** [comment]
closing this for now, feel free to re-open
